L1TriggerConfig/Utilities/src/L1TMuonOverlapFwVersionTester.cc

```diff
@@ -0,0 +1,66 @@
+#include <iomanip>
+#include <iostream>
+
+#include "FWCore/Framework/interface/EDAnalyzer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+
+#include "CondFormats/DataRecord/interface/L1TMuonOverlapFwVersionRcd.h"
+#include "CondFormats/DataRecord/interface/L1TMuonOverlapFwVersionO2ORcd.h"
+#include "CondFormats/L1TObjects/interface/L1TMuonOverlapFwVersion.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "CondCore/DBOutputService/interface/PoolDBOutputService.h"
+#include "CondCore/CondDB/interface/Session.h"
+
+using namespace std;
+
+class L1TMuonOverlapFwVersionTester : public edm::EDAnalyzer {
+private:
+  bool isO2Opayload;
+  bool writeToDB;
+
+public:
+  void analyze(const edm::Event &, const edm::EventSetup &) override;
+
+  explicit L1TMuonOverlapFwVersionTester(const edm::ParameterSet &pset) : edm::EDAnalyzer() {
+    isO2Opayload = pset.getUntrackedParameter<bool>("isO2Opayload", false);
+    writeToDB = pset.getUntrackedParameter<bool>("writeToDB", false);
+  }
+  ~L1TMuonOverlapFwVersionTester(void) override {}
+};
+
+void L1TMuonOverlapFwVersionTester::analyze(const edm::Event &iEvent, const edm::EventSetup &evSetup) {
+  edm::ESHandle<L1TMuonOverlapFwVersion> handle1;
+  if (isO2Opayload)
+    evSetup.get<L1TMuonOverlapFwVersionO2ORcd>().get(handle1);
+  else
+    evSetup.get<L1TMuonOverlapFwVersionRcd>().get(handle1);
+  std::shared_ptr<L1TMuonOverlapFwVersion> ptr1(new L1TMuonOverlapFwVersion(*(handle1.product())));
```

Comment:
It looks like you have an additional copy that could be removed, for example by coding:
const L1TMuonOverlapFwVersion& obj = *handle1.product();
In this way, you can move the below call to writeOne to writeOneIOv(obj

Reply:
also I think the handle1 is not used for anything else then just to get the product of it, so one can use the getData() method of the ES, which currently is using a get() method and the handle. I pointed to the twiki to follow in https://github.com/cms-sw/cmssw/pull/35876/files/250d619bf5e78fa43e2b76a77b872ab6993d3d64#r747562872